src/composer/prover.rs

```diff
@@ -0,0 +1,511 @@
+use rand_core::{CryptoRng, RngCore};
+
+use crate::commitment_scheme::CommitKey;
+use crate::error::Error;
+use crate::fft::{EvaluationDomain, Polynomial as FftPolynomial};
+use crate::proof_system::proof::Proof;
+use crate::proof_system::{
+    linearization_poly, quotient_poly, ProverKey, VerifierKey,
+};
+use crate::transcript::TranscriptProtocol;
+use crate::util;
+
+use super::{Builder, Circuit, Composer};
+
+/// Turbo Prover with processed keys
+#[derive(Clone)]
+pub struct Prover<C>
+where
+    C: Circuit,
+{
+    label: Vec<u8>,
+    pub(crate) prover_key: ProverKey,
+    pub(crate) commit_key: CommitKey,
+    pub(crate) verifier_key: VerifierKey,
+    pub(crate) transcript: Transcript,
+    pub(crate) size: usize,
+    pub(crate) constraints: usize,
+    circuit: PhantomData<C>,
+}
+
+impl<C> ops::Deref for Prover<C>
+where
+    C: Circuit,
+{
+    type Target = ProverKey;
+
+    fn deref(&self) -> &Self::Target {
+        &self.prover_key
+    }
+}
+
+impl<C> Prover<C>
+where
+    C: Circuit,
+{
+    pub(crate) fn new(
+        label: Vec<u8>,
+        prover_key: ProverKey,
+        commit_key: CommitKey,
+        verifier_key: VerifierKey,
+        size: usize,
+        constraints: usize,
+    ) -> Self {
+        let transcript =
+            Transcript::base(label.as_slice(), &verifier_key, constraints);
+
+        Self {
+            label,
+            prover_key,
+            commit_key,
+            verifier_key,
+            transcript,
+            size,
+            constraints,
+            circuit: PhantomData,
+        }
+    }
+
+    /// adds blinding scalars to a witness vector
+    ///
+    /// appends:
+    ///
+    /// if hiding degree = 1: (b2*X^(n+1) + b1*X^n - b2*X - b1) + witnesses
+    /// if hiding degree = 2: (b3*X^(n+2) + b2*X^(n+1) + b1*X^n - b3*X^2 - b2*X
+    fn blind_poly<R>(
+        rng: &mut R,
+        witnesses: &[BlsScalar],
+        hiding_degree: usize,
+        domain: &EvaluationDomain,
+    ) -> FftPolynomial
+    where
+        R: RngCore + CryptoRng,
+    {
+        let mut w_vec_inverse = domain.ifft(witnesses);
+
+        for i in 0..hiding_degree + 1 {
+            let blinding_scalar = util::random_scalar(rng);
+
+            w_vec_inverse[i] = w_vec_inverse[i] - blinding_scalar;
+            w_vec_inverse.push(blinding_scalar);
+        }
+
+        FftPolynomial::from_coefficients_vec(w_vec_inverse)
+    }
+
+    fn prepare_serialize(
+        &self,
+    ) -> (usize, Vec<u8>, Vec<u8>, [u8; VerifierKey::SIZE]) {
+        let prover_key = self.prover_key.to_var_bytes();
+        let commit_key = self.commit_key.to_raw_var_bytes();
+        let verifier_key = self.verifier_key.to_bytes();
+
+        let label_len = self.label.len();
+        let prover_key_len = prover_key.len();
+        let commit_key_len = commit_key.len();
+        let verifier_key_len = verifier_key.len();
+
+        let size =
+            48 + label_len + prover_key_len + commit_key_len + verifier_key_len;
```

Comment:
magic 48

Reply:
Its the fixed size of the lengths that defines the arbitrary bytes. But indeed should be a documented constant. Fixed!